src/functionaltests/java/com/ericsson/ei/subscriptions/trigger/SubscriptionTriggerSteps.java

```diff
@@ -1,44 +1,245 @@
+        LOGGER.debug("Verifying REST requests.");
+        assert(requestBodyContainsStatedValues(new JSONArray(mockClient.retrieveRecordedRequests(request().withPath(REST_ENDPOINT), Format.JSON))));
+        assert(requestBodyContainsStatedValues(new JSONArray(mockClient.retrieveRecordedRequests(request().withPath(REST_ENDPOINT_AUTH), Format.JSON))));
+        assert(requestBodyContainsStatedValues(new JSONArray(mockClient.retrieveRecordedRequests(request().withPath(REST_ENDPOINT_PARAMS), Format.JSON))));
+        assert(requestBodyContainsStatedValues(new JSONArray(mockClient.retrieveRecordedRequests(request().withPath(REST_ENDPOINT_AUTH_PARAMS), Format.JSON))));
+    }
 
+    private void readSubscriptionNames(String jsonDataAsString) throws Throwable {
+        JSONArray jsonArray = new JSONArray(jsonDataAsString);
+        for (int i = 0; i < jsonArray.length(); i++) {
+            subscriptionNames.add(jsonArray.getJSONObject(i).get("subscriptionName").toString());
+        }
     }
 
+    private void postSubscriptions(String jsonDataAsString, String endPoint) throws Exception {
+        RequestBuilder requestBuilder = MockMvcRequestBuilders.post(endPoint).accept(MediaType.APPLICATION_JSON)
+                .content(jsonDataAsString).contentType(MediaType.APPLICATION_JSON);
+
+        postResult = mockMvc.perform(requestBuilder).andReturn();
+        LOGGER.debug("Response code from REST when adding subscriptions: "
+                + String.valueOf(postResult.getResponse().getStatus()));
+
+        assertEquals(HttpStatus.OK.value(), postResult.getResponse().getStatus());
+    } 
+
+    private void validateSubscriptionsSuccessfullyAdded(String endPoint) throws Exception {
+        RequestBuilder getRequest = MockMvcRequestBuilders.get(endPoint);
+        getResult = mockMvc.perform(getRequest).andReturn();
+
+        LOGGER.debug("Response code from REST when getting subscriptions: "
+                + String.valueOf(getResult.getResponse().getStatus()));
+        assertEquals(HttpStatus.OK.value(), result.getResponse().getStatus());
+
+        LOGGER.debug("Checking that response contains all subscriptions");
+        for (String subscriptionName : subscriptionNames) {
+            assertTrue(getResult.getResponse().getContentAsString().contains(subscriptionName));
+        }        
+    }
+
+    private boolean requestBodyContainsStatedValues(JSONArray jsonArray) throws JSONException {
+        int tc5 = 0, successfull = 0;
+        for(int i = 0; i < jsonArray.length(); i++){
+            String requestBody = jsonArray.getString(i);
+            if (requestBody.contains("TC5")) {
+                tc5++;
+            }
+            if (requestBody.contains("SUCCESSFUL")) {
+                successfull++;
+            }
+        }
+        return (tc5 > 0 && successfull > 0);
+    }
+
+    private void setupRestEndpoints() {
+        // Set up endpoints
+        int port = SocketUtils.findAvailableTcpPort();
```

Comment:
This could maybe be put in the FunctionalTestBase class, and sending in the desired endpoints, so other tests can reuse it.

Reply:
The method is not general but specific to our test, if to make it general then you basucly replace the one line "mockClient.when(request().withMethod("POST").withPath(REST_ENDPOINT)).respond(response().withStatusCode(201));" with 4 calls to the method, this may not reduce the amount of code needed. And other tests may want other reponse codes, and maybe get and not post and so on.